packages/react-server/src/ReactFlightServer.js

```diff
@@ -1836,6 +1835,15 @@ function serializeLimitedObject(): string {
   return '$Y';
 }
 
+function serializeHaltedReference(): string {
```

Comment:
Same point here. This is not a halted reference. This is a blocked reference that was generated by halting but might be generated other ways too.